### PR TITLE
[controller] Fixed Store config JSON encoding

### PR DESCRIPTION
**Pull request by FelixGV**
We used to have a rmdVersionId field in the Store config, with a type of Optional<Integer>, but this did not get encoded properly by Jackson.

Starting in this commit, in order to avoid breaking on previously-encoded data, we are renaming the field to rmdVersion, and changing its type to just int, with -1 indicating that it is unset.

Also renamed the config: replication.metadata.version.id to replication.metadata.version so that it is consistent with the code.

Alternative way of resolving #294.

## How was this PR tested?
New unit test to ensure that the old (badly-encoded data) didn't break the new way of doing it. Also wrote a unit for the reverse (rollback from new way to old way) but it is not part of the commit since supporting that in the test suite would require carrying lots of dead code...

## Does this PR introduce any user-facing changes?
- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

Config rename, and ZK property rename.

### CODE REVIEW

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java [hunk 1]
```diff
@@ -2423,7 +2423,7 @@ public void updateStore(String clusterName, String storeName, UpdateStoreQueryPa
               .orElseGet(currStore::isWriteComputationEnabled);
       setStore.replicationMetadataVersionID = replicationMetadataVersionID
           .map(addToUpdatedConfigList(updatedConfigsList, REPLICATION_METADATA_PROTOCOL_VERSION_ID))
-          .orElse(currStore.getRmdVersionID().orElse(-1));
+          .orElse(currStore.getRmdVersion());
```
Comment (eolivelli) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java:2426]: in this case the value is  0 and not -1 (because zero is the default value)

**Review by nisargthakkar — DISMISSED** [review]

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java (hunk 1 above)
Comment (nisargthakkar) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java:2426]: `Store#getRmdVersionID` from all code paths was ending up in `ZkStore#getRmdVersionID` which had the logic:
```
@Override
  public Optional<Integer> getRmdVersionID() {
    final int rmdVersionID = this.storeProperties.replicationMetadataVersionID;
    return rmdVersionID == -1 ? Optional.empty() : Optional.of(rmdVersionID);
  }
```

Since we changed 
```

**Review by sixpluszero — DISMISSED** [review]

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java (hunk 1 above)
Comment (FelixGV) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java:2427]: I think what Enrico is indicating is that Jackson might default to 0 for an int property that isn't present in the payload. That may be true in general (IDK) but I verified that it is not the case in this particular implementation because in the `ZkStore` constructor we do:

```
this.storeProperties = Store.prefillAvroRecordWithDefaultValue(new StoreProperties());
```

And so the property of interest ends up getting filled with Avro's default (-1 in this case).

I added a test to assert this in the latest commit. Also modified another test to ensure that when the controller finds -1 in this store-level config, it will default to the controller-level config.

So in summary, I think the implementation in the original commit was already correct, but I've tweaked the unit tests to ensure it, and to ensure we catch regressions in those areas.

Thanks!

**Comment by FelixGV** [comment]
Thanks a lot Nisarg!

**Comment by eolivelli** [comment]
This patch works in my environment. Nice work!
